### ISSUE TITLE
Error is `json` data files are in working directory

### ISSUE THREAD

**Issue by ethanwhite** [weecology/retriever#637]
If there are `json` data files in the working directory this causes the following error:

```
ethan@croryx:~/RetrieverCode (master)$ retriever ls
Traceback (most recent call last):
  File "/home/ethan/anaconda/bin/retriever", line 9, in <module>
    load_entry_point('retriever===1.9.0-de', 'console_scripts', 'retriever')()
  File "/home/ethan/anaconda/lib/python2.7/site-packages/setuptools-20.2.2-py2.7.egg/pkg_resources/__init__.py", line 549, in load_entry_point
  File "/home/ethan/anaconda/lib/python2.7/site-packages/setuptools-20.2.2-py2.7.egg/pkg_resources/__init__.py", line 2542, in load_entry_point
  File "/home/ethan/anaconda/lib/python2.7/site-packages/setuptools-20.2.2-py2.7.egg/pkg_resources/__init__.py", line 2202, in load
  File "/home/ethan/anaconda/lib/python2.7/site-packages/setuptools-20.2.2-py2.7.egg/pkg_resources/__init__.py", line 2208, in resolve
  File "/home/ethan/anaconda/lib/python2.7/site-packages/retriever/__main__.py", line 29, in <module>
    from retriever.lib.datapackage import create_json, edit_json
  File "/home/ethan/anaconda/lib/python2.7/site-packages/retriever/lib/datapackage.py", line 8, in <module>
    short_names = [script.shortname.lower() for script in SCRIPT_LIST()]
  File "/home/ethan/anaconda/lib/python2.7/site-packages/retriever/__init__.py", line 102, in SCRIPT_LIST
    return [module.SCRIPT for module in MODULE_LIST(force_compile)]
  File "/home/ethan/anaconda/lib/python2.7/site-packages/retriever/__init__.py", line 78, in MODULE_LIST
    compile_json(join(search_path, script_name))
  File "/home/ethan/anaconda/lib/python2.7/site-packages/retriever/lib/compile.py", line 205, in compile_json
    json_object = json.load(open(json_file + ".json", "r"))
  File "/home/ethan/anaconda/lib/python2.7/json/__init__.py", line 291, in load
    **kw)
  File "/home/ethan/anaconda/lib/python2.7/json/__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "/home/ethan/anaconda/lib/python2.7/json/decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/ethan/anaconda/lib/python2.7/json/decoder.py", line 380, in raw_decode
    obj, end = self.scan_once(s, idx)
ValueError: Expecting object: line 1 column 1 (char 0)
```

@henrykironde already has a plan for fixing this, but I just wanted to flag it.

**Comment by ethanwhite** [comment]
A fix was added for this in 5bed7a4340248abf8cf3139933eef970689b97ec, but it seems to just produce a new problem now:

```
ethan@gandalf:~/RetrieverCode (master)$ retriever ls
Traceback (most recent call last):
  File "/home/ethan/anaconda/bin/retriever", line 9, in <module>
    load_entry_point('retriever==2.0.dev0', 'console_scripts', 'retriever')()
  File "/home/ethan/anaconda/lib/python3.5/site-packages/setuptools-23.0.0-py3.5.egg/pkg_resources/__init__.py", line 542, in load_entry_point
  File "/home/ethan/anaconda/lib/python3.5/site-packages/setuptools-23.0.0-py3.5.egg/pkg_resources/__init__.py", line 2569, in load_entry_point
  File "/home/ethan/anaconda/lib/python3.5/site-packages/setuptools-23.0.0-py3.5.egg/pkg_resources/__init__.py", line 2229, in load
  File "/home/ethan/anaconda/lib/python3.5/site-packages/setuptools-23.0.0-py3.5.egg/pkg_resources/__init__.py", line 2235, in resolve
  File "/home/ethan/anaconda/lib/python3.5/site-packages/retriever-2.0.dev0-py3.5.egg/retriever/__main__.py", line 29, in <module>
    from retriever.lib.datapackage import create_json, edit_json
  File "/home/ethan/anaconda/lib/python3.5/site-packages/retriever-2.0.dev0-py3.5.egg/retriever/lib/datapackage.py", line 8, in <module>
    short_names = [script.shortname.lower() for script in SCRIPT_LIST()]
  File "/home/ethan/anaconda/lib/python3.5/site-packages/retriever-2.0.dev0-py3.5.egg/retriever/__init__.py", line 111, in SCRIPT_LIST
    return [module.SCRIPT for module in MODULE_LIST(force_compile)]
  File "/home/ethan/anaconda/lib/python3.5/site-packages/retriever-2.0.dev0-py3.5.egg/retriever/__init__.py", line 78, in MODULE_LIST
    compile_json(join(search_path, script_name))
  File "/home/ethan/anaconda/lib/python3.5/site-packages/retriever-2.0.dev0-py3.5.egg/retriever/lib/compile.py", line 90, in compile_json
    if "retriever" not in json_object.keys():
AttributeError: 'list' object has no attribute 'keys'
```

Both Python 2 and 3.

**Comment by zhangcandrew** [comment]
I'm not sure if this has been resolved, but I looked a little into this. Here is the chunk of code from where the error triggers:

```
try:
        json_object = json.load(open(json_file + ".json", "r"))
    except ValueError as e:
        pass
    if type(json_object) is not dict:
        return
    if "retriever" not in json_object.keys():
        # Compile only files that have retriever key
        return
```

from lib/compile.py. 

I'm not 100% sure why the error triggers, since the list should be caught in the boolean before it calls for object.keys(), but I think we can circumvent this problem by using the python function hasattr(). We could place it as another check, inline with the check to see if the json_object is a dictionary. 


Also, I seem to be unable to replicate this error. Let me know what you think.

**Comment by ethanwhite** [comment]
Thanks for looking into this @zhangcandrew. I think @henrykironde fixed this a few days ago in 3b34ce698a1eb1daa67886e34a8606bc094f0406 and we just didn't get the issue closed.

@henrykironde - if I'm understanding that right from the history can you close this issue.

**Comment by henrykironde** [comment]
Yes, this is fixed. I that is why you don't get error again.
